frontend/src/metabase/query_builder/components/NativeQueryEditor.jsx

```diff
@@ -384,7 +397,8 @@ class NativeQueryEditor extends Component {
     // e.g. if `|` is the cursor position and the user is typing:
     //   - {{#123-foo|}} will fetch completions for the word "123-foo"
     //   - {{#123 foo|}} will not fetch completions because the word "foo" is not the first word in the tag.
-    if (prefix !== this.getCardSlugAtCursor(pos)) {
+    // Note we need to drop the leading `#` from the card tag name because the prefix only includes alphanumerics
+    if (prefix !== this.getCardTagNameAtCursor(pos).substring(1)) {
```

Comment:
Note all the changes in this file preserve behaviour and are just tidying things up